enhancements/20230327-harvester-seeder-integration.md

```diff
@@ -0,0 +1,244 @@
+
+sample secret creation call to endpoint: `${ENDPOINT}/v1/harvester/secrets`
+```json
+{
+  "type": "Opaque",
+  "metadata": {
+    "namespace": "harvester-system",
+    "name": "harvester-659jw"
+  },
+  "_type": "Opaque",
+  "data": {
+    "password": "TjBJTE80dSEh",
+    "username": "RXRob3M="
+  }
+}
+```
+
+sample inventory creation call to endpoint: `${ENDPOINT}/v1/harvester/metal.harvesterhci.io.inventories`
+
+```json
+{
+  "kind": "Inventory",
+  "metadata": {
+    "name": "harvester-659jw",
+    "namespace": "harvester-system",
+    "annotations": {
+      "metal.harvesterhci.io/localInventory": "true",
+      "metal.harvesterhci.io/localInventoryStatus": "eyJvd25lckNsdXN0ZXIiOiB7Im5hbWUiOiAibG9jYWwiLCJuYW1lc3BhY2UiOiAiaGFydmVzdGVyLXN5c3RlbSJ9LCJweGVCb290Q29uZmlnIjogeyJhZGRyZXNzIjogIjE3Mi4xOS4xMDguMTAiLCAiZ2F0ZXdheSI6IiIsICJuZXRtYXNrIjoiIn0sInN0YXR1cyI6ICJpbnZlbnRvcnlOb2RlUmVhZHkiLCAiZ2VuZXJhdGVkUGFzc3dvcmQiOiIiLCAiaGFyZHdhcmVJRCI6IiJ9Cg=="
+    }
+  },
+  "spec": {
+    "baseboardSpec": {
+      "connection": {
+        "authSecretRef": {
+          "name": "harvester-659jw",
+          "namespace": "harvester-system"
+        },
+        "host": "172.19.1.172",
+        "insecureTLS": true,
+        "port": 623
+      }
+    },
+    "events": {
+      "enabled": true,
+      "pollingInterval": "1h"
+    }
+  }
+}
+```
+
+Since no actual provisioning is performed by seeder, a base64 encoded status for inventory object is passed in the object using the annotation key `metal.harvesterhci.io/localInventoryStatus`. The address for corresponding node must match the address in `.pxeBootConfig.address`, as this is used to match inventory with corresponding node in the seeder cluster reconciliation. This is to ensure that the hardware discovery and hardware event information is correlated with the correct Harvester node.
+
+```json
+{
+  "ownerCluster": {
+    "name": "local",
+    "namespace": "harvester-system"
+  },
+  "pxeBootConfig": {
+    "address": "172.19.109.10",
+    "gateway": "",
+    "netmask": ""
+  },
+  "status": "inventoryNodeReady",
+  "generatedPassword": "",
+  "hardwareID": ""
+}
+
+```
+
+Once the inventory object is created, the `local cluster controller`, will add the inventory to the `local` cluster object in the `harvester-system` namespace.
+
+The `cluster event controller` will now regularly reconcile inventory objects and query underlying hardware for hardware details and events, and trigger updates to the harvester nodes.
+
+The `local node controller`, will also watch nodes for power action annotations `metal.harvesterhci.io/actionRequested` and trigger appropriate power actions by leveraging the rufio machine, rufio task and rufio job controllers.
+
+Once the associated job is completed, the `metal.harvesterhci.io/actionRequested` annotation will be removed, and replaced with 
+`metal.harvesterhci.io/actionStatus` which will indicate status of the action.
```

Comment:
ditto, would adding a new spec and status be more straightforward?

Reply:
will move the power actions to inventory status, as this can then be used in multi cluster mode as well to trigger adhoc power cycles.